### PR TITLE
Fix Memo program ESM exports

### PR DESCRIPTION
**Pull request by jordansexton**
This is the first of two PRs to address https://github.com/solana-labs/solana-program-library/issues/3502, this one for the Memo program JS (which the Token program JS depends on). This fixes the issues that build systems have with .mjs files by exporting .js files. It also restructures the project to be similar to Wallet Adapter / the TypeScript monorepo template I've been using. I'll update Token in parallel.

### CODE REVIEW

File: memo/js/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+root = true
+
+[*]
+charset = utf-8
+indent_style = space
+indent_size = 4
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
```
Comment (joncinque) [inline memo/js/.editorconfig:9]: Can you remove this?

File: memo/js/package.json [hunk 2]
```diff
@@ -1,70 +1,77 @@
...
-    "typescript": "^4.7.3",
-    "typescript-esm": "^2.0.0"
-  },
-  "dependencies": {
-    "@solana/web3.js": "^1.41.0",
-    "buffer": "^6.0.3"
-  }
+    "name": "@solana/spl-memo",
+    "description": "SPL Memo Program JS API",
+    "version": "0.3.1",
```
Comment (joncinque) [inline memo/js/package.json:4]: Not a big deal, but 0.3.0 should probably be next -- was this version copy-pasted from Token?

File: memo/js/.editorconfig (hunk 1 above)
Comment (jordansexton) [reply memo/js/.editorconfig:9]: I can, but the value of it is that it makes an IDE use the same spacing settings as Prettier is configured to here. Thoughts?

File: memo/js/package.json (hunk 2 above)
Comment (jordansexton) [reply memo/js/package.json:4]: Oh, you're right. This should be 0.2.1, this is a nonbreaking change.

**Comment by jordansexton** [comment]
> Why is there no lockfile at all now?

Fixed!

File: memo/js/.editorconfig (hunk 1 above)
Comment (joncinque) [reply memo/js/.editorconfig:9]: It's just spacing, so I'd prefer not.  Many IDEs infer from prettier files what to do as well. Finally, CI checks for it anyway, so let's axe it

Comment (jordansexton) [reply memo/js/.editorconfig]: Got it. Removed!

**Comment by jordansexton** [comment]
Published in @solana/spl-memo@0.2.1

**Pull request merged**